gef-ui/server/controllers/amendments/cover-end-date/validation.test.ts

```diff
@@ -0,0 +1,93 @@
+import { applyStandardValidationAndParseDateInput, COVER_END_DATE_MAXIMUM_YEARS_IN_FUTURE } from '@ukef/dtfs2-common';
+import { add, startOfDay, getUnixTime } from 'date-fns';
+import { validateAndParseCoverEndDate } from './validation';
+import { mapValidationError } from '../../../utils/map-validation-error';
+
+const valueName = 'cover end date';
+const valueRef = 'coverEndDate';
+
+describe('validateAndParseCoverEndDate', () => {
+  afterAll(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should map the response from applyStandardValidationAndParseDateInput', () => {
+    // Arrange
+    const dayMonthYear = {
+      day: 'x',
+      year: '2024',
+      month: '1',
+    };
+
+    // Act
+    const response = validateAndParseCoverEndDate(dayMonthYear, new Date());
+
+    // Assert
+    expect(response).toEqual({
+      errors: [mapValidationError(applyStandardValidationAndParseDateInput(dayMonthYear, valueName, valueRef).error!)],
+    });
+  });
+
+  it('should return error if date is before cover start date', () => {
+    // Arrange
+    const coverEndDate = new Date();
+    const coverStartDate = add(new Date(), { days: 1 });
+
+    // Act
+    const result = validateAndParseCoverEndDate(
+      { day: coverEndDate.getDate().toString(), month: (coverEndDate.getMonth() + 1).toString(), year: coverEndDate.getFullYear().toString() },
+      coverStartDate,
```

Comment:
I think its more consistent to use `format` from date-fns. This gets rid of the issue of months being 0 indexed
```suggestion
      { day: format(coverEndDate, 'd'), month: format(coverEndDate, 'M'), year: format(coverEndDate, 'yyyy') },
      coverStartDate,
```